doc/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# (C) Copyright 2019-2020 UCAR.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+################################################################################
+# IODA-CONVERTER fixed files
+################################################################################
+
+list( APPEND bufr_obs_samples
+  bufr_obs_samples/1bamua.bufr.subset.txt
+  bufr_obs_samples/adpupa.bufr.subset.txt
+  bufr_obs_samples/adpupa.prepbufr.subset.txt
+  bufr_obs_samples/aircraft.bufr.subset.txt
+  bufr_obs_samples/aircraft.prepbufr.subset.txt
+  bufr_obs_samples/aod.bufr.subset.txt
+  bufr_obs_samples/gpsro.bufr.subset.txt
+)
+
+list( APPEND bufr_table_samples
+  bufr_table_samples/1bamua.bufr.table.txt
+  bufr_table_samples/adpupa.bufr.table.txt
+  bufr_table_samples/adpupa.prepbufr.table.txt
+  bufr_table_samples/aircraft.bufr.table.txt
+  bufr_table_samples/aircraft.prepbufr.table.txt
+  bufr_table_samples/aod.bufr.table.txt
+  bufr_table_samples/gpsro.bufr.table.txt
+)
+
+list( APPEND fix
+  fix/ghcn-stations.txt
+)
+
+# create doc directories and make links to the doc fix files
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bufr_obs_samples)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bufr_table_samples)
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/fix)
+foreach(FILENAME ${bufr_obs_samples} ${bufr_table_samples} ${fix})
+    execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink
+           ${CMAKE_CURRENT_SOURCE_DIR}/${FILENAME}
+           ${CMAKE_CURRENT_BINARY_DIR}/${FILENAME} )
+endforeach(FILENAME)
```

Comment:
This configuration takes care of accessing these files from the build area for testing, but is lacking the commands to have these files installed (when `make install` is run). I'm assuming that it is essential to have the ghcn-stations.txt file installed for the script to be able to run from the installation area.

You want to insert a cmake install command at the end that looks like:

```
install( FILES ${fix} DESTINATION doc )
```

This should install the list of files to `${CMAKE_INSTALL_PREFIX}/doc` which in this case results in `${CMAKE_INSTALL_PREFIX}/doc/fix/ghcn-stations.txt`.

The bufr samples are optional but it might be useful to install these as well. If you want to install everything, you could do:

```
list( APPEND all_files
  ${fix}
  ${bufr_obs_samples}
  ${bufr_table_samples}
}

install( FILES ${all_files} DESTINATION doc )
```

Reply:
As suggested. I applied your install everything suggestion and  add the lines you suggested in the end  of doc/CMakeLists.txt  and commit and push to this feature directory.

Reply:
When I add them to doc/CMakeLists.txt, ecbuild fails due to configuration. Look this suggestion does not work. The error message is "-- The following OPTIONAL packages have not been found:

 * oops
 * ioda_engines
 * bufr

-- Configuring incomplete, errors occurred!

Reply:
@YoulongXia-NOAA thanks for working on this. I'll take a look at the code and help debug.

Reply:
I found issue as you write as 

list( APPEND all_files
  ${fix}
  ${bufr_obs_samples}
  ${bufr_table_samples}
}
 for both email and github.  I directly copy it and paste to my doc/CMakeLists.txt and save it. Now I found the last one is ) rather than }. Now it works.  Thank you for your suggestion.

Reply:
Thanks for letting me know. I just found that error too. Sorry about having the syntax error in the suggestion I sent you.

I tested this in my local clone and this works, but doesn't preserve the directory structure from the source area. Ie, after installing the fix, bufr_obs_samples and bufr_table_samples directories are missing and all the file copied to doc. I trying out a different form of the install command that should preserve the directory structure.

Reply:
Got it working to preserve the directory structure under the doc directory when installing. I'll approve now, but wait for the tests to finish to merge.